src/content-script.ts

```diff
@@ -4,14 +4,47 @@ import Mustache from "mustache";
 import { createElement } from "./elements";
 import { isBodyWithReplies, isRepositoryDetails } from "./validations";
 
-(async function main() {
+const regexPathName = /\/.+\/.+\/(issues|pull)\/\d+/;
+const replySummaryId = "saved-reply-new_comment_field";
+const footprint = "refined-saved-replies";
+
+// hack for detecting page change
+function addObserver(callback: MutationCallback) {
+	const options = { childList: true, subtree: true };
+
+	const observer = new MutationObserver(callback);
```

Comment:
[Cleanup?] It's good practice (I think) to `observer.disconnect()` when it no longer needs to be fired. The `footprint` already kind of does work to prevent duplicate actions, but not having the observer at all would be even cleaner.

E.g.

```ts
new MutationObserver(() => {
    callback();
    observer.disconnect();
});
```

I wonder if the `footprint` would still be necessary at all if the observer disconnects? Or rather, maybe it'd be possible to get away with not having the footprint?

E.g.

```plaintext
on content script run:
   if page matches:
       main
   else
       set up observer
```

Reply:
OH I just re-realized - we might need to run `main()` multiple times! SO I think this comment thread is bogus and I'm wrong and you shouldn't listen to me :smile:. Keeping it here for posterity (at least I will forget why you/we did this, and perhaps others will wonder too).